src/commands/basic/genbanlist.js

```diff
@@ -0,0 +1,55 @@
+const fetch = require("node-fetch")
+const { apiurl } = require("../../../config.json")
+const ConfigModel = require("../../database/schemas/config")
+const { MessageEmbed } = require("discord.js");
+const Str = require("@supercharge/strings")
+const fs = require("fs")
+
+module.exports = {
+    config: {
+        name: "genbanlist",
+        aliases: [],
+        usage: "",
+        category: "basic",
+        description: "Creates a .json banlist file to use for servers",
+        accessibility: "Moderator",
+    },
+    run: async (client, message, args) => {
+        const config = await ConfigModel.findOne({guildid: message.guild.id})
+        if (!config.trustedCommunities) return message.reply("Please set trusted communities first")
+        if (!config.ruleFilters) return message.reply("Please set rule filters first")
+        message.reply("Processing banlist. Please wait")
+
+        let trustedCommunitiesProm = config.trustedCommunities.map(id => {
+            return fetch(`${apiurl}/communities/getid?id=${id}`).then(res => res.text())
+        })
+        const trustedCommunities = await Promise.all(trustedCommunitiesProm)
```

Comment:
I assume that you are talking about `trustedCommunitiesProm` not being used. That is because it is an array of promises that resolve to an array of violations for every community.

Reply:
My first line was suggesting to start the rule fetchs before you await the community fetches.
My second line was talking about `trustedCommunities` not `trustedCommunitiesProm` and I assume you were ment to use `trustedCommunities` during the fetch to rules, which would invalidate my first line.

Reply:
Ah. That was because I forgot to implement only trusted communities being in the banlist, I added it now. Didn't test it properly due to the banlist being one long line instead of a nicely structured JSON file. I also fixed the issue of it not being a nice string has also been resolves, it's a nicely formatted JSON file now that is much more human-readable.